### DISCUSSION TITLE
Unit Testing

### DISCUSSION THREAD

**Discussion by ranjitrn5** [Netflix/dgs-framework#319] [Q&A]
I am trying to write a unit test for one of my DataFetcher in which I am using DgsContext to get the request Header. An example is shown below in my code:

`String code= DgsContext.getRequestData(dfe).getHeaders().get("Authorization").get(0);`

Now during writing the unit test am unable to write a unit test for this since it seems DgsContext has a static anonymous function it seems?

Either ways would there be any documentation on how can I get past this. Currently in my test it throws a NullPOinterException since the test runner isnt aware about the DgsContext at all.

**Comment by sbilello** [comment]
https://github.com/Netflix/dgs-framework/pull/302/files try to use this approach
